### PR TITLE
support for Deeplake v4 api

### PR DESCRIPTION
**Pull request by azat-manukyan**
# Description
added support for deeplake version >= 4.0.0

### CODE REVIEW

File: llama-index-legacy/llama_index/legacy/vector_stores/deeplake.py
Comment (logan-markewich) [inline llama-index-legacy/llama_index/legacy/vector_stores/deeplake.py:1]: Tbh, legacy isn't really meant to be maintained -- its a copy of llama-index from v0.9.62, its meant to help people who didn't want to immediately migrate to v0.10.x, but its not going to be there forever

File: llama-index-integrations/vector_stores/llama-index-vector-stores-deeplake/pyproject.toml [hunk 1]
```diff
@@ -31,7 +31,7 @@ version = "0.2.2"
 
 [tool.poetry.dependencies]
 python = ">=3.9,<4.0"
-deeplake = ">=3.9.12,<4.0"
+deeplake = ">=3.9.12"
```
Comment (logan-markewich) [inline llama-index-integrations/vector_stores/llama-index-vector-stores-deeplake/pyproject.toml:34]: This change will work for both deeplake 3.x and 4.x?

File: llama-index-integrations/vector_stores/llama-index-vector-stores-deeplake/pyproject.toml
Comment (logan-markewich) [inline llama-index-integrations/vector_stores/llama-index-vector-stores-deeplake/pyproject.toml:1]: We can bump the version on the integration as well

File: llama-index-integrations/vector_stores/llama-index-vector-stores-deeplake/pyproject.toml (hunk 1 above)
Comment (logan-markewich) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-deeplake/pyproject.toml:34]: ah, I see the backward compat

**Review by logan-markewich — APPROVED** [review]

File: llama-index-integrations/vector_stores/llama-index-vector-stores-deeplake/pyproject.toml
Comment (logan-markewich) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-deeplake/pyproject.toml:1]: Actually I did it

File: llama-index-legacy/llama_index/legacy/vector_stores/deeplake.py
Comment (azat-manukyan) [reply llama-index-legacy/llama_index/legacy/vector_stores/deeplake.py:1]: thanks for your comments. will not do further improvements inside the legacy folder

**Review by logan-markewich — APPROVED** [review]

**Pull request merged**